src/lib.ts

```diff
@@ -485,18 +489,82 @@ export function userConfirmDialog(msg: string, defaultValue: boolean, timeOut?:
   return userConfirm(msg);
 }
 
-export const latteActionSourceFinderConstraints = {
-  allowedAction: (action: ActionSource): boolean => {
-    if (!have($item`latte lovers member's mug`)) return true;
-    const forceEquipsOtherThanLatte = (
-      action?.constraints?.equipmentRequirements?.().maximizeOptions.forceEquip ?? []
-    ).filter((equipment) => equipment !== $item`latte lovers member's mug`);
-    return (
-      forceEquipsOtherThanLatte.every((equipment) => toSlot(equipment) !== $slot`off-hand`) &&
-      sum(forceEquipsOtherThanLatte, weaponHands) < 2
-    );
-  },
-};
+function determineFreeBunnyBanish(): boolean {
+  const extraOrbFights = have($item`miniature crystal ball`) ? 1 : 0;
+  const possibleGregsFromSpleen =
+    Math.floor((spleenLimit() - mySpleenUse()) / 2) * (3 + extraOrbFights);
+  const currentAvailableGregs = get("beGregariousCharges") * (3 + extraOrbFights);
+  const habitatFights = (3 - get("_monsterHabitatsRecalled")) * (5 + extraOrbFights);
```

Comment:
I think it'd be good to clamp these values between 0 and 3 in case a user sets them to values outside the normal range.